helper/validate_password.go

```diff
@@ -35,44 +39,36 @@ func NewPasswordValidator() PasswordValidator {
 	if validator != nil {
 		return validator
 	}
-	dictionaryPath := os.Getenv("PASS_DICTIONARY_PATH") // path from user
+	path := os.Getenv(dictionaryPath) // path from user
 	validator := &passwordValidator{
 		validator: crunchy.NewValidatorWithOpts(crunchy.Options{
-			DictionaryPath: dictionaryPath, // is path is empty, crunchy will use default value  "/usr/share/dict"
+			DictionaryPath:    path, // if the path is empty, crunchy will use the default value  "/usr/share/dict"
+			MinLength:         minPasswordLengths,
+			MustContainDigit:  true,
+			MustContainSymbol: true,
 		}),
 	}
 	return validator
 }
 
 func (p *passwordValidator) Validate(pass string) error {
-
-	if len(pass) < minPasswordLengths {
-		return ErrLength
+	rate, err := p.validator.Rate(pass)
+	if err != nil {
+		if errors.Is(err, crunchy.ErrTooShort) {
+			return ErrLength // return more user-friendly message
+		}
+		return err
 	}
-
-	uppercase := regexp.MustCompile(`[A-Z]`)
-	if !uppercase.MatchString(pass) {
+	if rate > minPasswordRate {
+		return nil
+	}
+	// crunchy doesn't return err if the pass doesn't contain upper/lower case letter
+	// we need to check the pass to return a more user-friendly message
+	if !uppercasePattern.MatchString(pass) {
 		return ErrUpper
 	}
-
-	lowercase := regexp.MustCompile(`[a-z]`)
-	if !lowercase.MatchString(pass) {
+	if !lowercasePattern.MatchString(pass) {
 		return ErrLow
 	}
-
-	number := regexp.MustCompile(`\d`)
-	if !number.MatchString(pass) {
-		return ErrNum
-	}
-
-	specialChar := regexp.MustCompile(`[@$!%*?&]`)
-	if !specialChar.MatchString(pass) {
-		return ErrSymbol
-	}
-
-	err := p.validator.Check(pass)
-	if err != nil {
-		return fmt.Errorf("weak pass: %s", err)
-	}
-	return nil
+	return ErrWeak
```

Comment:
why you return ErrWeak here instead of nil?

Reply:
I did it to decrease `if` nested
in another case it will be
```go
	if rate < minPasswordRate {
		if !uppercasePattern.MatchString(pass) {
			return ErrUpper
		}
		if !lowercasePattern.MatchString(pass) {
			return ErrLow
		}
		return ErrWeak
	}
	return nil
```

Reply:
I see
may I ask you to use provided sample above
IMHO it is a bit more explicit
to decrease complexity you may use small functions to incapsulate IFs, e.g.
```
rate, err := validatorRate(pass)
if err != nil {
	return err
}

if rate < minPasswordRate {
	return handlePwdLowRateError(...)
}

return nil

OR

rate, err := validatorRate(pass)
if err != nil {
	return err
}

return handlePwdRate(rate)
```
maybe @arbrix could suggest more sophisticated way to do that

Reply:
Done